tools/internal_ci/linux/grpc_distribtests_php.sh

```diff
@@ -45,6 +48,8 @@ cp -r artifacts/* input_artifacts/ || true
 # a better signal about which distribtest are affected by the currently broken artifact builds.
 tools/run_tests/task_runner.py -f distribtest linux php -j 4 -x distribtests/sponge_log.xml || FAILED="true"
 
+tools/internal_ci/helper_scripts/store_artifacts_from_moved_src_tree.sh
```

Comment:
This is a distribtest, do we need to preserve artifacts?

Reply:
Yes, since is a flow of "build artifacts locally and then run distribtests on them". Since we've built the artifacts, it's useful to store them.
Actually one day we might not need the universal "grpc_build_artifacts" jobs because all the artifacts would be build by these per-language jobs (and they would immediately be tested with the distribtests). This would be future work of course, for now we'll still need the grpc_build_artifacts flow (since it's tightly bound with our release process).

 Same for other "grpc_distribtest_LANGUAGE" jobs.

Reply:
You mentioned Ruby artifacts diverge between build_artifacts and distribtests. If it's potentially useful in future, and an  immediate optimization. Maybe we can optimize now? Up to you.